wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+#!/usr/bin/env python3
+from __future__ import print_function, division
+import h5py
+import numpy as np
+import sys
+import csv
+import os
+import ROOT
+from ROOT import gROOT
+from ROOT import TMath
+
+# ROOT.ROOT.EnableThreadSafety()
+
+# DEFAULT VALUES FOR SPE FIT:
+D_OFFSET = 0
+D_LAMBDA = 1
+D_SPE_CHARGE = 0.8
+D_NOISE_SPREAD = 0.01
+D_SPE_CHARGE_SPREAD = 0
+D_ZERO_PEAK_SPREAD = 0.4
+# Number of peaks we use for SPE fitting
+n = 20
+
+def iqr(h):
+    """
+    Returns the iqr of the histogram `h`
+    """
+    l = None
+    r = None
+    total = h.GetEntries()
+    i = 0
+    n = h.GetBinContent(i)
+    while n / total < 0.25:
+        i += 1
+        n += h.GetBinContent(i)
+    high_prob = n / total
+    n -= h.GetBinContent(i)
+    i -= 1
+    low_prob = n / total
+    l = (((0.25 - low_prob) / (high_prob - low_prob)) * h.GetBinWidth(i)) + h.GetBinLowEdge(i)
+    while n / total < 0.75:
+        i += 1
+        n += h.GetBinContent(i)
+    high_prob = n / total
+    n -= h.GetBinContent(i)
+    i -= 1
+    low_prob = n / total
+    r = (((0.75 - low_prob) / (high_prob - low_prob)) * h.GetBinWidth(i)) + h.GetBinLowEdge(i)
+    return r-l
+
+def average_distance_from(val, h):
+    """
+    Standard deviation calculation, except use `val` instead of the mean.
+    """
+    d2 = 0
+    for i in range(h.GetMaximumBin() + 2):
+        d2 += (val - h.GetXaxis().GetBinCenter(i))**2 * h.GetBinContent(i)
+    return np.sqrt(d2)
```

Comment:
Is this used anywhere? I think also `h` needs to be normalized for this to work, otherwise you have to use `h.GetBinContent(i)/h.Sum()` or something like that.

Reply:
Yeah this isn't used anymore. Deleted.